plugins/janus_videoroom.c

```diff
@@ -4408,21 +4441,17 @@ static void janus_videoroom_hangup_media_internal(janus_plugin_session *handle)
 		participant->remb_latest = 0;
 		participant->fir_latest = 0;
 		participant->fir_seq = 0;
-		while(participant->subscribers) {
-			janus_videoroom_subscriber *s = (janus_videoroom_subscriber *)participant->subscribers->data;
+		GSList *subscribers = participant->subscribers;
+		participant->subscribers = NULL;
+		janus_mutex_unlock(&participant->subscribers_mutex);
```

Comment:
See note above: this unlock was *after* the while before. Although I guess it's pretty much the same as here we're working on a copy of the subscribers list, and the method itself removes from the *actual* list using the mutex?